### PR TITLE
Update phpunit for php 7.2 compatible test runs

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| License | MIT

#### What's in this PR?

Update phpunit ^6.5 with matthiasnoback config-test and dependency-test.

Set travis and appveyor to php 7.2

Following replaces done:

Update namespaces:

```
PHPUnit\Framework\
PHPUnit_Framework_

fix use PHPUnit\Framework\Assert

extends PHPUnit\Framework
extends \PHPUnit\Framework
```

Move TestCase to use:

```
(\nclass([\s\S^?]+?) extends )(\\PHPUnit\\Framework\\)
use $3TestCase;\n$1
```

Fix comments before TestCase classes.

Update getMock functions:

```
getMock\((.*?)\)
getMockBuilder($1)->getMock()

->getMockBuilder()->getMock()
->getMock()
```

Update `@expectException`

```
setExpectedException
expectException

\n(     \* )@expectExceptionMessage (.*?)(\n)([\s\S^?]+?)(\{)
$3$4$5\n        \$this->expectExceptionMessage(\'$2\');

\n(     \* )@expectException (.*?)(\n)([\s\S^?]+?)(\{)
$3$4$5\n        \$this->expectException($2::class);
```

Replace stdClass:

```
$this->getMockBuilder('stdClass', 
$this->getMockBuilder('TestClass')->setMethods(
```

#### Why?

Newer version of phpunit is needed for 7.2

#### Example Usage

~~~php
bin/runtests
~~~

### TODO

 - [x] Fix PHP 7.2 https://travis-ci.org/sulu/sulu/jobs/371339802
 - [x] Fix PHP 7.1 https://travis-ci.org/sulu/sulu/jobs/371339803
 - [x] Fix risky tests

### CODE REVIEW

**Comment by alexander-schranz** [comment]
What should we do with the risky tests: https://travis-ci.org/sulu/sulu/jobs/371292997?

File: src/Sulu/Bundle/SecurityBundle/Tests/Functional/Command/CreateUserCommandTest.php [hunk 1]
```diff
@@ -74,11 +74,11 @@ public function testCreateUserNonExistingLocale()
     }
 
     /**
-     * @expectedException \RuntimeException
-     * @expectedExceptionMessage The system currently has no roles. Use the "sulu:security:role:create" command to create roles.
+     * @expectExceptionMessage The system currently has no roles. Use the "sulu:security:role:create" command to create roles.
```
Comment (danrot) [inline src/Sulu/Bundle/SecurityBundle/Tests/Functional/Command/CreateUserCommandTest.php]: Shouldn't we move that into the test itself as well?

File: src/Sulu/Bundle/MediaBundle/Tests/Unit/Media/FormatLoader/XmlFormatLoader10Test.php [hunk 2]
```diff
@@ -132,11 +132,9 @@ public function testLoadDeprecated()
         $this->assertNotNull($result['3840x2160-retina']['internal']);
     }
 
-    /**
-     * @expectedException \Sulu\Bundle\MediaBundle\Media\FormatLoader\Exception\MissingScaleDimensionException
-     */
     public function testLoadDeprecatedWithMissingDimension()
     {
+        $this->expectException(\Sulu\Bundle\MediaBundle\Media\FormatLoader\Exception\MissingScaleDimensionException::class);
```
Comment (danrot) [inline src/Sulu/Bundle/MediaBundle/Tests/Unit/Media/FormatLoader/XmlFormatLoader10Test.php]: According to our styleguides this class should actually be included via a `use` statement on top. And actually the same is true for all the `TestCase` classes from PHPUnit... Isn't there some StyleCI or PhpCsFixer rule we could apply that automatically fixes all of those for us?

Comment (alexander-schranz) [reply src/Sulu/Bundle/MediaBundle/Tests/Unit/Media/FormatLoader/XmlFormatLoader10Test.php]: can not find anything about this :(

Comment (alexander-schranz) [reply src/Sulu/Bundle/MediaBundle/Tests/Unit/Media/FormatLoader/XmlFormatLoader10Test.php]: did go manually over the `expectException` functions and fix it there

**Comment by alexander-schranz** [comment]
@danrot risky tests should be fixed.

**Pull request merged**